### ISSUE TITLE
RefindPlus hangs on the main screen

### ISSUE THREAD

**Issue by startergo** [dakanji/RefindPlus#139]
### RefindPlus Version

Pre-Release Code Build

### Device Type

UEFI PC

### Problem Description

Upon boot RefindPlus boots to the main screen showing only the logo.

### Problem Point

RefindPlus fails to load

### Affected Items

All OS Loaders (First Row Items)

### Debug Log

No log provided as the debug version does not build with Docker.

### Additional Context

[RefindPlus.efi.zip](https://github.com/dakanji/RefindPlus/files/10311481/RefindPlus.efi.zip)

**Comment by dakanji** [comment]
You mean the 3rd party Docker process at https://github.com/xaionaro/edk2-builder-docker.

Sounds more like a problem that should be raised there. If it is not building the DBG file, it might be building a corrupt REL file.
If you have an issue after following the recommended build steps, this can be looked at here but FYI, that builds both DBG and REL without issue and I am booted via the REL file from that build right now.

Eitherway, something to fix on that project.

**Comment by startergo** [comment]
Have you tested the attached file? It may be working on your machine?

**Comment by dakanji** [comment]
That's a good point. It appears to be working for me indeed.

I suppose you still ultimately need to build with the provided build script and output logs before any forward steps can be taken.

**Comment by startergo** [comment]
Rebuilt with GCC5 and it works fine for the RELEASE version. For the DEBUG there are errors with GCC5:

<details>
  <summary>Error during build</summary>

```
Building ... /home/edk2/edk2/MdePkg/Library/BasePrintLib/BasePrintLib.inf [X64]
In file included from /home/edk2/edk2/OpenCorePkg/Library/OcDevicePathLib/OcDevicePathLib.c:15:
/home/edk2/edk2/OpenCorePkg/Library/OcDevicePathLib/../../../RefindPlusPkg/Library/OCRefitLib/OCRefitLib.h:60:5: error: unknown type name 'CHAR16'
   60 |     CHAR16 EFIAPI CharToUpper (IN CHAR16 Char);
      |     ^~~~~~
/home/edk2/edk2/OpenCorePkg/Library/OcDevicePathLib/../../../RefindPlusPkg/Library/OCRefitLib/OCRefitLib.h:60:19: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'CharToUpper'
   60 |     CHAR16 EFIAPI CharToUpper (IN CHAR16 Char);
      |                   ^~~~~~~~~~~
Building ... /home/edk2/edk2/MdePkg/Library/BaseDebugLibNull/BaseDebugLibNull.inf [X64]
"gcc" -Wno-error -Os -DEFIX64 -D__MAKEWITH_TIANO -DREFIT_DEBUG=1 -Wno-error -c -o /home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/OpenCorePkg/Library/OcGuardLib/OcGuardLib/OUTPUT/./Alignment.obj -I/home/edk2/edk2/OpenCorePkg/Library/OcGuardLib -I/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/OpenCorePkg/Library/OcGuardLib/OcGuardLib/DEBUG -I/home/edk2/edk2/MdePkg -I/home/edk2/edk2/MdePkg/Include -I/home/edk2/edk2/MdePkg/Include/X64 -I/home/edk2/edk2/OpenCorePkg -I/home/edk2/edk2/OpenCorePkg/Include/AMI -I/home/edk2/edk2/OpenCorePkg/Include/Acidanthera -I/home/edk2/edk2/OpenCorePkg/Include/Apple -I/home/edk2/edk2/OpenCorePkg/Include/Generic -I/home/edk2/edk2/OpenCorePkg/Include/Intel -I/home/edk2/edk2/OpenCorePkg/Include/Microsoft -I/home/edk2/edk2/OpenCorePkg/Include/VMware -I/home/edk2/edk2/OpenCorePkg/Include/Apple/X64 /home/edk2/edk2/OpenCorePkg/Library/OcGuardLib/Alignment.c
Building ... /home/edk2/edk2/MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf [X64]
"gcc" -Wno-error -Os -DEFIX64 -D__MAKEWITH_TIANO -DREFIT_DEBUG=1 -Wno-error -c -o /home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BasePcdLibNull/BasePcdLibNull/OUTPUT/./PcdLib.obj -I/home/edk2/edk2/MdePkg/Library/BasePcdLibNull -I/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BasePcdLibNull/BasePcdLibNull/DEBUG -I/home/edk2/edk2/MdePkg -I/home/edk2/edk2/MdePkg/Include -I/home/edk2/edk2/MdePkg/Include/X64 /home/edk2/edk2/MdePkg/Library/BasePcdLibNull/PcdLib.c
Building ... /home/edk2/edk2/MdePkg/Library/BaseLib/BaseLib.inf [X64]
"gcc" -Wno-error -Os -DEFIX64 -D__MAKEWITH_TIANO -DREFIT_DEBUG=1 -Wno-error -c -o /home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BasePrintLib/BasePrintLib/OUTPUT/./PrintLibInternal.obj -I/home/edk2/edk2/MdePkg/Library/BasePrintLib -I/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BasePrintLib/BasePrintLib/DEBUG -I/home/edk2/edk2/MdePkg -I/home/edk2/edk2/MdePkg/Include -I/home/edk2/edk2/MdePkg/Include/X64 /home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c
Building ... /home/edk2/edk2/MdePkg/Library/BaseMemoryLib/BaseMemoryLib.inf [X64]
"gcc" -Wno-error -Os -DEFIX64 -D__MAKEWITH_TIANO -DREFIT_DEBUG=1 -Wno-error -c -o /home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BaseDebugLibNull/BaseDebugLibNull/OUTPUT/./DebugLib.obj -I/home/edk2/edk2/MdePkg/Library/BaseDebugLibNull -I/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BaseDebugLibNull/BaseDebugLibNull/DEBUG -I/home/edk2/edk2/MdePkg -I/home/edk2/edk2/MdePkg/Include -I/home/edk2/edk2/MdePkg/Include/X64 /home/edk2/edk2/MdePkg/Library/BaseDebugLibNull/DebugLib.c
"gcc" -Wno-error -Os -DEFIX64 -D__MAKEWITH_TIANO -DREFIT_DEBUG=1 -Wno-error -c -o /home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib/OUTPUT/./DevicePathUtilities.obj -I/home/edk2/edk2/MdePkg/Library/UefiDevicePathLib -I/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib/DEBUG -I/home/edk2/edk2/MdePkg -I/home/edk2/edk2/MdePkg/Include -I/home/edk2/edk2/MdePkg/Include/X64 /home/edk2/edk2/MdePkg/Library/UefiDevicePathLib/DevicePathUtilities.c
In file included from /home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.h:22,
                 from /home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:15:
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c: In function 'BasePrintLibConvertValueToStringS':
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumAsciiStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:24:32: note: in expansion of macro 'PcdGet32'
   24 | #define ASCII_RSIZE_MAX       (PcdGet32 (PcdMaximumAsciiStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:377:9: note: in expansion of macro 'ASCII_RSIZE_MAX'
  377 |     if (ASCII_RSIZE_MAX != 0) {
      |         ^~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: note: each undeclared identifier is reported only once for each function it appears in
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:24:32: note: in expansion of macro 'PcdGet32'
   24 | #define ASCII_RSIZE_MAX       (PcdGet32 (PcdMaximumAsciiStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:377:9: note: in expansion of macro 'ASCII_RSIZE_MAX'
  377 |     if (ASCII_RSIZE_MAX != 0) {
      |         ^~~~~~~~~~~~~~~
/home/edk2/edk2/OpenCorePkg/Library/OcFileLib/FileProtocol.c: In function 'OcFindWritableFileSystem':
/home/edk2/edk2/OpenCorePkg/Library/OcFileLib/FileProtocol.c:229:7: warning: passing argument 3 of 'OcSafeFileOpen' from incompatible pointer type [-Wincompatible-pointer-types]
  229 |       L"octest.fil",
      |       ^~~~~~~~~~~~~
      |       |
      |       int *
In file included from /home/edk2/edk2/OpenCorePkg/Library/OcFileLib/FileProtocol.c:23:
/home/edk2/edk2/OpenCorePkg/Include/Acidanthera/Library/OcFileLib.h:116:32: note: expected 'const CHAR16 *' {aka 'const short unsigned int *'} but argument is of type 'int *'
  116 |   IN  CONST CHAR16            *FileName,
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumUnicodeStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:23:32: note: in expansion of macro 'PcdGet32'
   23 | #define RSIZE_MAX             (PcdGet32 (PcdMaximumUnicodeStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:384:9: note: in expansion of macro 'RSIZE_MAX'
  384 |     if (RSIZE_MAX != 0) {
      |         ^~~~~~~~~
/home/edk2/edk2/OpenCorePkg/Library/OcDevicePathLib/OcDevicePathLib.c: In function 'InternalFileDevicePathsEqualWorker':
/home/edk2/edk2/OpenCorePkg/Library/OcDevicePathLib/OcDevicePathLib.c:812:15: warning: implicit declaration of function 'CharToUpper' [-Wimplicit-function-declaration]
  812 |       Char1 = CharToUpper ((*FilePath1)->PathName[Clip1Index + Index]);
      |               ^~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c: In function 'BasePrintLibSPrintMarker':
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumUnicodeStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:23:32: note: in expansion of macro 'PcdGet32'
   23 | #define RSIZE_MAX             (PcdGet32 (PcdMaximumUnicodeStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:594:9: note: in expansion of macro 'RSIZE_MAX'
  594 |     if (RSIZE_MAX != 0) {
      |         ^~~~~~~~~
make: *** [GNUmakefile:350: /home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/OpenCorePkg/Library/OcDevicePathLib/OcDevicePathLib/OUTPUT/OcDevicePathLib.obj] Error 1
"gcc" -Wno-error -Os -DEFIX64 -D__MAKEWITH_TIANO -DREFIT_DEBUG=1 -Wno-error -c -o /home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BaseLib/BaseLib/OUTPUT/./CheckSum.obj -I/home/edk2/edk2/MdePkg/Library/BaseLib/X64 -I/home/edk2/edk2/MdePkg/Library/BaseLib -I/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BaseLib/BaseLib/DEBUG -I/home/edk2/edk2/MdePkg -I/home/edk2/edk2/MdePkg/Include -I/home/edk2/edk2/MdePkg/Include/X64 /home/edk2/edk2/MdePkg/Library/BaseLib/CheckSum.c
/home/edk2/edk2/MdePkg/Include/Library/PcdLib.h:388:45: error: '_PCD_GET_MODE_32_PcdMaximumAsciiStringLength' undeclared (first use in this function)
  388 | #define PcdGet32(TokenName)                 _PCD_GET_MODE_32_##TokenName
      |                                             ^~~~~~~~~~~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:24:32: note: in expansion of macro 'PcdGet32'
   24 | #define ASCII_RSIZE_MAX       (PcdGet32 (PcdMaximumAsciiStringLength))
      |                                ^~~~~~~~
/home/edk2/edk2/MdePkg/Library/BasePrintLib/PrintLibInternal.c:599:9: note: in expansion of macro 'ASCII_RSIZE_MAX'
  599 |     if (ASCII_RSIZE_MAX != 0) {
      |         ^~~~~~~~~~~~~~~


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/OpenCorePkg/Library/OcDevicePathLib/OcDevicePathLib]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BasePrintLib/BasePrintLib]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/OpenCorePkg/Library/OcGuardLib/OcGuardLib]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/OpenCorePkg/Library/OcFileLib/OcFileLib]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/Build/RefindPlus/DEBUG_GCC5/X64/MdePkg/Library/BaseLib/BaseLib]


build.py...
 : error F002: Failed to build module
	/home/edk2/edk2/OpenCorePkg/Library/OcDevicePathLib/OcDevicePathLib.inf [X64, GCC5, DEBUG]

- Failed -
Build end time: 11:57:05, Dec.28 2022
Build total time: 00:00:09
```

</details>

**Comment by dakanji** [comment]
Sorry, but I can only look into issues with building with the provided build script.

**Comment by startergo** [comment]
I was just laying out the results. Opened separate issue in the Docker repo.
